index.js

```diff
@@ -47,7 +64,7 @@ function co(gen) {
   // we wrap everything in a promise to avoid promise chaining,
   // which leads to memory leak errors.
   // see https://github.com/tj/co/issues/180
-  return new Promise(function(resolve, reject) {
+  return new coPromise(function(resolve, reject) {
```

Comment:
Are we sure this syntax will work for whatever promise library the user wants? If not, should there be some sort of whitelist, or do we want it to just throw the error it would normally throw if the promise library for whatever reason doesn't work (i.e. fail loudly, but without a specific error message saying "this promise library doesn't support this syntax").